### DISCUSSION TITLE
QuoteTableAndColumns setting not honoured when mapping IList<KeyValuePair> to a table

### DISCUSSION THREAD

**Discussion by bounav** [nhibernate/fluent-nhibernate#565] [Q&A] [answered]
I've recently discovered the `SchemaMetadataUpdater.QuoteTableAndColumns` setting when you configure fluent nhibernate.

This options tells nhibernate to escape all table and column names. For example a column named `Key` will be surrounded by _[_ and _]_ if you are using T-SQL or _`_  (backtick) if your database is mysql.

Example on how you could turn on the _Quote Table and Columns_ options.

```csharp
IPersistenceConfigurer configuration = MsSqlConfiguration.MsSql2012
                                                         .ConnectionString(connectionString)
                                                         .AdoNetBatchSize(1000);

var fluentConfig = Fluently.Configure()
                           .Database(configuration)
                           .ExposeConfiguration(SchemaMetadataUpdater.QuoteTableAndColumns)
                           .Mappings(m => m.FluentMappings.Add(fluentMappingTypes));
```

This setting seems to work fine until you try to map a `IList<KeyValuePair>` using the fluent table mapping as follow:

```csharp
// Domain object
public class Foo
{
    public int Id { get; set; }
    public string Name { get; set; }
    public IList<KeyValuePair> Parameters { get; set; }
}

// Mapping class
public class FooMap : ClassMap<Foo>
{
	public FooMap()
	{
		Id(x => x.Id).GeneratedBy.Assigned();
		Map(x => x.Name);
		HasMany(x => x.Parameters)
			.Table("FooParameter")
			.KeyColumn("Foo_id")
			.Component(c =>
			{
				// Without the Column name override trying to save an instance of Foo will fail
				c.Map(x => x.Key).ColumnName("[Key]"));
				c.Map(x => x.Value);
			})
			.Cascade.AllDeleteOrphan();
	}
}
```

Am I wrong to assume that the `c.Map(x => x.Key)` and `c.Map(x.Value)` should automatically quote the column names?

**Comment by hazzik** [comment, marked as answer]
I think this was fixed in around NH 5

**Reply by bounav** [reply]
It's been a while since this was originally asked ;). I checked in our project and we no longer have any trace of using `.ColumnName()` so I'll assume it has been fixed as you suggested.
